sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/ExtractKeyPhraseClient.java

```diff
@@ -0,0 +1,117 @@
+import com.azure.ai.textanalytics.implementation.models.DocumentError;
+import com.azure.ai.textanalytics.implementation.models.DocumentKeyPhrases;
+import com.azure.ai.textanalytics.implementation.models.KeyPhraseResult;
+import com.azure.ai.textanalytics.implementation.models.MultiLanguageBatchInput;
+import com.azure.ai.textanalytics.models.DocumentResultCollection;
+import com.azure.ai.textanalytics.models.ExtractKeyPhraseResult;
+import com.azure.ai.textanalytics.models.TextAnalyticsRequestOptions;
+import com.azure.ai.textanalytics.models.TextDocumentInput;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static com.azure.ai.textanalytics.Transforms.mapByIndex;
+import static com.azure.ai.textanalytics.Transforms.toBatchStatistics;
+import static com.azure.ai.textanalytics.Transforms.toMultiLanguageInput;
+import static com.azure.ai.textanalytics.Transforms.toTextAnalyticsError;
+import static com.azure.ai.textanalytics.Transforms.toTextDocumentStatistics;
+import static com.azure.core.util.FluxUtil.monoError;
+
+/**
+ * Helper class for managing extract keyphrase endpoint.
+ */
+class ExtractKeyPhraseClient {
+    private final ClientLogger logger;
+    private final TextAnalyticsClientImpl service;
+
+    /**
+     * Create a {@code ExtractKeyPhraseClient} that sends requests to the Text Analytics services's extract keyphrase
+     * endpoint.
+     *
+     * @param service The proxy service used to perform REST calls.
+     * @param logger The logger for the {@link TextAnalyticsAsyncClient} class.
+     */
+    ExtractKeyPhraseClient(TextAnalyticsClientImpl service, ClientLogger logger) {
+        this.logger = logger;
+        this.service = service;
+    }
+
+    Mono<Response<ExtractKeyPhraseResult>> extractKeyPhrasesWithResponse(String text, String language,
+        Context context) {
+        Objects.requireNonNull(text, "'text' cannot be null.");
+
+        return extractBatchKeyPhrasesWithResponse(
+            Collections.singletonList(new TextDocumentInput("0", text, language)), null, context)
+            .map(response -> new SimpleResponse<>(response, response.getValue().iterator().next()));
+    }
+
+    Mono<Response<DocumentResultCollection<ExtractKeyPhraseResult>>> extractKeyPhrasesWithResponse(
+        List<String> textInputs, String language, Context context) {
+        Objects.requireNonNull(textInputs, "'textInputs' cannot be null.");
+
+        List<TextDocumentInput> documentInputs = mapByIndex(textInputs, (index, value) ->
+            new TextDocumentInput(index, value, language));
+        try {
+            return extractBatchKeyPhrasesWithResponse(documentInputs, null, context);
+        } catch (RuntimeException ex) {
```

Comment:
This try-catch won't be necessary if the public async client wraps this method call with it's own try-catch block.

Reply:
This is required as this also being called in the sync client [here](https://github.com/Azure/azure-sdk-for-java/pull/7050/files#diff-609a9a6b185e83e28625f680f2c6a016R530) where it is not wrapped in the try-catch block.

Reply:
For sync client, returning mono error is not necessary. It's okay to just throw the exception.